lib/vtls/schannel.c

```diff
@@ -2049,15 +2187,20 @@ schannel_recv(struct Curl_easy *data, int sockindex,
   */
   if(len && !BACKEND->decdata_offset && BACKEND->recv_connection_closed &&
      !BACKEND->recv_sspi_close_notify) {
-    bool isWin2k = curlx_verify_windows_version(5, 0, PLATFORM_WINNT,
+
+#if defined(_WIN32_WINNT) && _WIN32_WINNT <= 0x0500
```

Comment:
I do not understand the whole thing of combining runtime and compile time checks like this. What is the purpose here?

Reply:
Removing dead code-paths (smaller compiler binary). This is inherited from the other PR that this PR depends on: https://github.com/curl/curl/pull/7727

Reply:
I don't think this is worth the effort as it over-complicates the code and makes reading it a lot harder. Generally speaking we try to perform Windows specific checks during runtime, because it is common for a Windows application to be build on a one platform version and then shipped to a couple of others. Therefore compile-time checks do not make much sense with the Windows ecosystem.